ProteoformSuiteInternal/Lollipop.cs

```diff
@@ -184,6 +189,103 @@ public void process_neucode_components(List<NeuCodePair> raw_neucode_pairs)
         #endregion DECONVOLUTION Public Fields
 
         #region DECONVOLUTION
+
+        public void promex_deconvolute(int maxcharge, int mincharge, int maxmass, int minmass)
+        {
+            foreach (InputFile f in input_files.Where(f => f.purpose == Purpose.RawFile)) {
+
+                Process proc = new Process();
+                ProcessStartInfo startInfo = new ProcessStartInfo();
+
+                string input = f.complete_path;
+                DirectoryInfo directory = new DirectoryInfo(Directory.GetCurrentDirectory());
+                string promexlocation = Path.Combine(directory.Parent.Parent.Parent.Parent.FullName, @"ProteoformSuiteInternal\ProMex");
+
+                startInfo.FileName = @"C:\WINDOWS\system32\cmd.exe";
+                startInfo.UseShellExecute = false;
+                startInfo.RedirectStandardInput = true;
+                startInfo.RedirectStandardOutput = false;
+                startInfo.CreateNoWindow = true;
+                proc.StartInfo = startInfo;
+
+                proc.Start();
+
+                proc.StandardInput.WriteLine("cd " + promexlocation);
+
+                proc.StandardInput.WriteLine("ProMex.exe -i " + input + " -minCharge " + mincharge + " -maxCharge " + maxcharge + " -minMass " + minmass + " -maxMass " + maxmass + " -csv y -maxThreads 0");
+                proc.StandardInput.Close();
+                proc.WaitForExit();
+
+
+                var type = Envelope.DeconType.ProMex;
+                double minMass = 0;
+                int numChargesRequired = 3;
+                string dir = Directory.GetCurrentDirectory();
+                Loaders.LoadElements(dir + @"\elements.dat");
+                var rawFile = ThermoDynamicData.InitiateDynamicConnection(f.complete_path);
+                Dictionary<int, Feature> featureIdToFeature = new Dictionary<int, Feature>();
+                StreamReader reader = new StreamReader(Path.Combine(Path.GetDirectoryName(f.complete_path), Path.GetFileNameWithoutExtension(f.complete_path)) + "_ms1ft.csv");
+                string line;
+                int lineNum = 1;
+
+                while (reader.Peek() > 0)
+                {
+                    line = reader.ReadLine();
+
+                    if (lineNum != 1)
+                    {
+                        var massWithScan = new Envelope(line, type);
+                        massWithScan.retentionTime = rawFile.GetOneBasedScan(massWithScan.scan_num).RetentionTime;
```

Comment:
add if statement if(massWithScan.retention time *is in correct range*) then do all of the stuff below.